src/lib/common/string.cpp

```diff
@@ -345,6 +405,21 @@ bool parseUrl(const std::string& url, std::string& host, int& port, std::string&
     }
   }
 
+  //
+  // Is 'port' a valid number?
+  // Remember the port is really an 'unsigned short', so numbers over MAXSHORT (0xFFFF) are
+  // not valid. Negative port numbers cannot exist and 0 is reserved.
+  //
+  if ((port > MAXSHORT) || (port <= 0))
```

Comment:
Zero is reserved. Fine.

If MAX_PORT exists then use it, but if doesn't exist, please define it and use it ;)